src/core/public/core_system.test.ts

```diff
@@ -17,118 +17,72 @@
-    this.start = jest.fn().mockReturnValue(mockInjectedMetadataStart);
-    return this;
-  }
-);
+const MockInjectedMetadataService = injectedMetadataServiceMock.create();
+const InjectedMetadataServiceConstructor = jest
+  .fn()
+  .mockImplementation(() => MockInjectedMetadataService);
 jest.mock('./injected_metadata', () => ({
-  InjectedMetadataService: MockInjectedMetadataService,
+  InjectedMetadataService: InjectedMetadataServiceConstructor,
 }));
 
-const mockFatalErrorsStart = {};
-const MockFatalErrorsService = jest.fn<FatalErrorsService, any>(function _MockFatalErrorsService(
-  this: any
-) {
-  this.start = jest.fn().mockReturnValue(mockFatalErrorsStart);
-  this.add = jest.fn();
-  return this;
-});
+const MockFatalErrorsService = fatalErrorsServiceMock.create();
+const FatalErrorsServiceConstructor = jest.fn().mockImplementation(() => MockFatalErrorsService);
 jest.mock('./fatal_errors', () => ({
-  FatalErrorsService: MockFatalErrorsService,
+  FatalErrorsService: FatalErrorsServiceConstructor,
 }));
 
-const mockI18nStart = {};
-const MockI18nService = jest.fn<I18nService, any>(function _MockI18nService(this: any) {
-  this.start = jest.fn().mockReturnValue(mockI18nStart);
-  this.stop = jest.fn();
-  return this;
-});
+const MockI18nService = i18nServiceMock.create();
+const I18nServiceConstructor = jest.fn().mockImplementation(() => MockI18nService);
 jest.mock('./i18n', () => ({
-  I18nService: MockI18nService,
+  I18nService: I18nServiceConstructor,
 }));
 
-const mockNotificationStart = {};
-const MockNotificationsService = jest.fn<NotificationsService, any>(
-  function _MockNotificationsService(this: any) {
-    this.start = jest.fn().mockReturnValue(mockNotificationStart);
-    this.add = jest.fn();
-    this.stop = jest.fn();
-    return this;
-  }
-);
-
+const MockNotificationsService = notificationServiceMock.create();
+const NotificationServiceConstructor = jest.fn().mockImplementation(() => MockNotificationsService);
 jest.mock('./notifications', () => ({
-  NotificationsService: MockNotificationsService,
+  NotificationsService: NotificationServiceConstructor,
 }));
 
-const mockHttp = {};
-const MockHttpService = jest.fn<HttpService, any>(function _MockNotificationsService(this: any) {
-  this.start = jest.fn().mockReturnValue(mockHttp);
-  this.stop = jest.fn();
-  return this;
-});
+const MockHttpService = httpServiceMock.create();
+const HttpServiceConstructor = jest.fn().mockImplementation(() => MockHttpService);
 jest.mock('./http', () => ({
-  HttpService: MockHttpService,
+  HttpService: HttpServiceConstructor,
 }));
 
-const mockBasePathStart = {};
-const MockBasePathService = jest.fn<BasePathService, any>(function _MockNotificationsService(
-  this: any
-) {
-  this.start = jest.fn().mockReturnValue(mockBasePathStart);
-  return this;
-});
+const MockBasePathService = basePathServiceMock.create();
+const BasePathServiceConstructor = jest.fn().mockImplementation(() => MockBasePathService);
```

Comment:
I think it's fine not to have `*Constructor` as a part of `*.mock.ts`. It's not widely used and it's easy to declare right at the place of use.